src/renderer/html_handlebars/helpers/navigation.rs

```diff
@@ -1,169 +1,81 @@
 use std::path::Path;
-use std::collections::BTreeMap;
 
 use serde_json;
 use handlebars::{Context, Handlebars, Helper, RenderContext, RenderError, Renderable};
 
+fn find_chapter<'a>(
+    chapter: &'a serde_json::Value,
+    needle: &str)
+    -> Result<Option<&'a serde_json::Map<String, serde_json::Value>>, RenderError> {
+    let chapter =
+        chapter.as_object()
+               .ok_or_else(|| RenderError::new("Chapter is not an object"))?;
```

Comment:
Do you mean `find_chapter` should silently ignore if `chapter` is not an object or if `chapter.children` is not an array?

Reply:
My thinking was that you're trying to find a chapter called `needle` in some JSON value. If the value isn't an object then as far as the function is concerned it can't find our needle. 

What would we be able to do if `chapter` wasn't an object (or `chapter.children` isn't an array) besides halting with an error?

Reply:
I think we should halt if `find_chapter` discovers a type error at runtime. Just returning `None` might make an issue with node creating harder to discover and debug.

Reply:
I like that reasoning :+1:

 I guess you raise the point that it'd be better if we could work with something a bit more strongly typed than a miscellaneous JSON value so those sorts of type errors are impossible. That's something for a later PR though.